base/sim/FairMCApplication.cxx

```diff
@@ -515,6 +516,9 @@ void FairMCApplication::InitOnWorker()
 
     mtx.lock();
     fRootManager->InitSink();
+    RegisterOutput();
+    fRootManager->WriteFolder();
+    mtx.unlock();
```

Comment:
Consider using `std::lock_guard`, please. It unlocks the mutex automatically when leaving the block (even in case of errors or exceptions).

```suggestion
    {
        std::lock_guard guard(mtx);
        fRootManager->InitSink();
        RegisterOutput();
        fRootManager->WriteFolder();
    }
```